File: src/Modelo/Placa.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Modelo;


public class Placa {
    private int id_placa;
    private String linea;
    private String color;
    private String compuesto;
    private int precio;
    private boolean beta;
    private int ancho;
    private int largo;


    public Placa() {
    }
    

    public int getLargo() {
        return largo;
    }

    public void setLargo(int largo) {
        this.largo = largo;
    }

    public int getId_placa() {
        return id_placa;
    }

    public void setId_placa(int id_placa) {
        this.id_placa = id_placa;
    }

    public String getLinea() {
        return linea;
    }

    public void setLinea(String linea) {
        this.linea = linea;
    }

    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getCompuesto() {
        return compuesto;
    }

    public void setCompuesto(String compuesto) {
        this.compuesto = compuesto;
    }

    public int getPrecio() {
        return precio;
    }

    public void setPrecio(int precio) {
        this.precio = precio;
    }

    public boolean isBeta() {
        return beta;
    }

    public void setBeta(boolean beta) {
        this.beta = beta;
    }

    public int getAncho() {
        return ancho;
    }

    public void setAncho(int ancho) {
        this.ancho = ancho;
    }
    
}

File: Modelo/Placa.cs
using System;

namespace Woodic.Modelo
{
    public class Placa
    {
        public int IdPlaca { get; set; }
        public string Linea { get; set; } = string.Empty;
        public string Compuesto { get; set; } = string.Empty;
        public string Color { get; set; } = string.Empty;
        public double Precio { get; set; }
        public bool Beta { get; set; }
        public string Proveedor { get; set; } = string.Empty;
        public int Ancho { get; set; } = 1830;
        public int Largo { get; set; } = 2400;

        public string BetaTexto => Beta ? "Con Veta" : "Sin Veta";
        public string DimensionesTexto => $"{Ancho} x {Largo}";
        public string PrecioFormateado => $"${Precio:N2}";

        public Placa() { }

        // Compatibilidad con código original
        public int getId_placa() => IdPlaca;
        public void setId_placa(int id) => IdPlaca = id;

        public string getLinea() => Linea;
        public void setLinea(string l) => Linea = l;

        public string getColor() => Color;
        public void setColor(string c) => Color = c;

        public string getCompuesto() => Compuesto;
        public void setCompuesto(string c) => Compuesto = c;

        public int getPrecio() => (int)Precio;
        public void setPrecio(int p) => Precio = p;

        public bool isBeta() => Beta;
        public void setBeta(bool b) => Beta = b;

        public int getAncho() => Ancho;
        public void setAncho(int a) => Ancho = a;

        public int getLargo() => Largo;
        public void setLargo(int l) => Largo = l;
    }
}
